Automattic/wp-calypso
# Performance Profiler: Improvements on the header

Fixes [Performance Profiler: Create header area for loading screen & result page](https://github.com/Automattic/dotcom-forge/issues/8660)
Follow-up of [Performance Profiler: Create header area for loading screen & result page](https://github.com/Automattic/wp-calypso/pull/93451)

## Proposed Changes

Improvements for the header area of site profiler as follows:

* Update header component folder structure for consistency
* Update styles as font color, font size and spacing
* Isolate the badge on its own component
* Add rounded out border on Tabs
* Update the navbar layout
* Update the logic to change the tab type without losing any additional parameter
* Properly display big URLs
* Add a flag to allow to hide the navigation tabs

## Why are these changes being made?

To match the layout proposed on Figma


## Testing Instructions

Same as https://github.com/Automattic/wp-calypso/pull/93451
Plus:

* Go to `/speed-test-tool?url=http://wordpress.com` (without the `tab` param) and check the site still working and the mobile tab is selected
* Go to `/speed-test-tool?url=http://wordpress.com&extra=1` and check after changing the tab to desktop the `extra` parameter still present
* Check the URL parameter with big URLS (only the last segment should be shown)
* Check the URL with a big URL and only one segment (without additional `\` on the pathname) and it should add ellipsis at the end)

| Short URL  | Big URL (several segments) | Big URL (one segment)
| ------------- | ------------- | ------------- |
|![CleanShot 2024-08-12 at 16 59 32@2x](https://github.com/user-attachments/assets/27e3286a-de70-4dc6-a5d6-e77f94d47522)|![CleanShot 2024-08-12 at 17 05 17@2x](https://github.com/user-attachments/assets/f8642b5a-7f6f-4a1b-af28-3ece6e2dc7ae)|![CleanShot 2024-08-12 at 17 30 16@2x](https://github.com/user-attachments/assets/2c7bec42-dc90-4986-9c75-3928fe55becd)
